### PR TITLE
Better selects

### PR DESCRIPTION
**Pull request by ajb**
@jrubenoff: here's the JS for the styled selects. Note that I've commented out everything to do with the `width` option, because I didn't want to pre-determine how you'll implement that. However, it's worth noting that in our current implementation, we have options for setting the width of the select at `full`, `auto`, or `[X]px`, and I'm pretty sure we're using each of those options.

We should aim to use the existing dropdown styles here, right?

This should also be easier now that I've updated the preview app to use Rails. Hoorah!

### CODE REVIEW

File: vendor/assets/stylesheets/dvl/core/forms.scss [hunk 1]
```diff
@@ -329,38 +339,92 @@ form {
...
+      }
+    }
+    &.error + div {
+      border-color: $errorColor;
+    }
+  }
+}
+
+.styled_select {
+  height: $styledSelectHeight;
```
Comment (dobthound) [inline vendor/assets/stylesheets/dvl/core/forms.scss]: Rule sets should be ordered as follows: @extends, @includes without @content, properties, @includes with @content, nested rule sets

File: vendor/assets/stylesheets/dvl/core/forms.scss [hunk 2]
```diff
@@ -329,38 +339,92 @@ form {
...
+//     -webkit-appearance: none;
+//     -moz-appearance: none;
+//     text-indent: 0.01px;
+//     text-overflow: '';
+//   }
+// }
+
+$styledSelectHeight: 2.25rem;
+
+.styled_select_wrapper {
```
Comment (dobthound) [inline vendor/assets/stylesheets/dvl/core/forms.scss]: Selector `styled_select_wrapper` should be written in lowercase with hyphens

File: vendor/assets/stylesheets/dvl/core/forms.scss [hunk 3]
```diff
@@ -329,38 +339,92 @@ form {
...
+        color: $darkBlue;
+      }
+    }
+    &.error + div {
+      border-color: $errorColor;
+    }
+  }
+}
+
+.styled_select {
```
Comment (dobthound) [inline vendor/assets/stylesheets/dvl/core/forms.scss]: Selector `styled_select` should be written in lowercase with hyphens

**Comment by ajb** [comment]
@jrubenoff: can you look again, and browser test too? all of the relevant use cases should now be covered in the preview.

**Comment by jrubenoff** [comment]
Looks good in Safari / Chrome / FF, will download an IE9 VM and test by sometime tomorrow night.

**Comment by ajb** [comment]
Why don't you use Browserstack? If you need help getting it setup, we can
do it today.

On Wed, Jan 7, 2015 at 10:31 PM, Josh Rubenoff <notifications@github.com>
wrote:

> Looks good in Safari / Chrome / FF, will download an IE9 VM and test by
> sometime tomorrow night.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dobtco/dvl-core/pull/22#issuecomment-69141533>.
>



-- 
Adam Becker
510.928.9111
@AdamJacobBecker

**Comment by ajb** [comment]
I would also recommend BrowserStack's screenshots feature -- I'll post an example on the checkboxes/radio thread.

Also, assuming that our goal is to make the regular `<select>` and the `dropdown_select` look the same, one potential issue is that the `dropdown_select` doesn't have an `focus` state -- I'm not sure if that's possible to achieve or not.

**Comment by jrubenoff** [comment]
Ah, I had no idea they let you test locally... thanks!

File: vendor/assets/stylesheets/dvl/core/forms.scss [hunk 4]
```diff
@@ -366,38 +370,57 @@ form {
...
 
-.select_wrapper {
-  max-width: 100%;
-  height: 28px;
-  background-color: #fff;
-  border: 1px solid $gray;
-  border-radius: $radius;
+$styledSelectHeight: 2.25rem;
+
+.styled_select_wrapper {
```
Comment (dobthound) [inline vendor/assets/stylesheets/dvl/core/forms.scss]: Selector `styled_select_wrapper` should be written in lowercase with hyphens

**Comment by jrubenoff** [comment]
Alrighty, should be good to go.

**Pull request merged**